Upgrade 3.0 migration guide to talk about removal of Go Apply<TypeName> (#5641)
pulumi/docs: content/docs/get-started/install/migrating-3.0.md

Then run `go mod download`

> ### Removing use of Apply<TypeName>

Reviewer: ```suggestion
### Go SDK removes Apply<TypeName>
```

In order to improve the performance of our Go SDK, we have removed:

* Apply<TypeName>
* Apply
* ApplyWithContext

We now suggest that you use:

* ApplyT
* ApplyTWithContext

```go
//before
containerDef := image.ImageName.ApplyString(func(name string) (string, error) {
    return name, nil
})

//after
containerDef := image.ImageName.ApplyT(func(name string) (string, error) {
    return name, nil
}).(pulumi.StringOutput)
```

> Notice that the difference would be casting the result of the ApplyT to the Output type

Reviewer: ```suggestion
When using ApplyT, remember to cast the result to the Output type
```
